integrationtest/mapper/pojo-base/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/IndexedBaseIT.java

```diff
@@ -449,11 +449,12 @@ public void error_indexedWithoutEntityMetadata() {
 				.isInstanceOf( SearchException.class )
 				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
 						.typeContext( IndexedWithoutEntityMetadata.class.getName() )
-						.failure(
-								"Type '" + IndexedWithoutEntityMetadata.class.getName()
-										+ "' is not marked as an entity type and is not abstract, yet it is indexed or targeted"
-										+ " by an association from an indexed type. Please check your configuration."
-						)
+						.failure( "Unable to index type '" + IndexedWithoutEntityMetadata.class.getName()
+								+ "': this type is not an entity type."
+								+ " If you only expect subtypes to be instantiated, make this type abstract."
+								+ " If you expect this exact type to be instantiated and want it to be indexed, make it an entity type."
+								+ " Otherwise, ensure this type and its subtypes are never indexed by removing the @Indexed annotation"
+								+ " or by annotating the type with @Indexed(enabled = false)." )
```

Comment:
It's a completely valid solution when you're in this situation:

```
@Entity
@Indexed
public class MyEntity {
  // ...
}

@MapperSuperclass
public classs MySuperclass extends MyEntity {
  // ...
}

@Entity
public class MySubEntity extends MySuperclass {
  // ...
}
```

When you have this hierarchy, `MySuperclass` must be abstract, and if you don't want to make it abstract you need to annotate it with `@Indexed(enabled = false)`.

Since it's very specific I'd rather not go into details in the exception message, but I think users should be aware that `@Indexed(enabled = false)` exists.